CMakeLists.txt

```diff
@@ -185,12 +185,12 @@ if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
     exec_program(/usr/bin/sw_vers ARGS
         -productVersion OUTPUT_VARIABLE MACOSX_VERSION_RAW)
     string(REGEX REPLACE
-        "([0-9]+)(?:\\.([0-9]+).*)*" "\\1"
+        "([0-9]+)(\\.([0-9]+).*)*" "\\1"
         MACOSX_MAJOR_VERSION
         "${MACOSX_VERSION_RAW}")
 
     string(REGEX REPLACE
-        "([0-9]+)(?:\\.([0-9]+).*)*" "\\2"
+        "([0-9]+)\\.([0-9]+)(\\.([0-9]+).*)*" "\\2"
```

Comment:
Use `"([0-9]+)(\\.([0-9]+).*)*" "\\3"` instead. Maybe it will help in version string like 10

Reply:
Maybe I misunderstood the original regex---I thought the idea here was to capture the second (minor) version number.  So wouldn't it suffice to split out the second `([0-9]+)` capture group and use that, and leave the `(\\.([0-9]+).*)*` to capture (and leave unused) any additional terms in the version number?

Reply:
I think the purpose of this regex is to get the second (minor) version number if it exists. Before this fix, we used forced major version 10 and then a minor number. But after that fix, we also took care of major version 11. Now, I don't remember whether it would not break if we didn't include cases with no minor number like "11".

Reply:
Thanks for the explanation; fixed in 28be9dea83.  I tested this and it seems to work fine. :+1: